File: skitchen_frontend/src/services/paymentService.js
import api from "./api";

export const listPayments = async ({ page = 1, limit = 20, orderId } = {}) => {
  const params = { page, limit };
  if (orderId) params.order_id = orderId;
  const res = await api.get("/payments", { params });
  return res.data;
};

export const createPayment = async ({ order_id, amount, method, status }) => {
  const res = await api.post("/payments", { order_id, amount, method, status });
  return res.data;
};

export const getPayment = async (id) => {
  const res = await api.get(`/payments/${id}`);
  return res.data;
};

export const updatePaymentStatus = async (id, status) => {
  const res = await api.put(`/payments/${id}/status`, { status });
  return res.data;
};

export const getReceiptUrl = (id) => {
  const rawBase = import.meta.env.VITE_API_BASE_URL || "http://localhost:3000/api";
  const base = rawBase.endsWith("/api") ? rawBase.slice(0, -4) : rawBase;
  return `${base}/api/payments/${id}/receipt`;
};

export const getOrderReceiptUrl = (orderId) => {
  const rawBase = import.meta.env.VITE_API_BASE_URL || "http://localhost:3000/api";
  const base = rawBase.endsWith("/api") ? rawBase.slice(0, -4) : rawBase;
  return `${base}/api/orders/${orderId}/receipt`;
};

File: mobile_app/lib/services/payment_service.dart
import 'package:mobile_app/services/api_client.dart';
import 'package:mobile_app/services/auth_storage.dart';

class PaymentService {
  PaymentService({ApiClient? apiClient}) : _api = apiClient ?? ApiClient();

  final ApiClient _api;

  Future<List<Map<String, dynamic>>> listPayments({int page = 1, int limit = 20}) async {
    final token = await AuthStorage.getToken();
    if (token == null || token.isEmpty) {
      throw Exception('Missing token');
    }

    final response = await _api.get(
      '/payments?page=$page&limit=$limit',
      headers: {
        'Authorization': 'Bearer $token',
      },
    );

    final data = response['data'];
    if (data is List) {
      return data
          .whereType<Map<String, dynamic>>()
          .map((e) => Map<String, dynamic>.from(e))
          .toList();
    }

    throw Exception('Invalid payments response');
  }

  Future<Map<String, dynamic>> getPayment(String id) async {
    final token = await AuthStorage.getToken();
    if (token == null || token.isEmpty) {
      throw Exception('Missing token');
    }

    final response = await _api.get(
      '/payments/$id',
      headers: {
        'Authorization': 'Bearer $token',
      },
    );

    final data = response['data'];
    if (data is Map<String, dynamic>) {
      return Map<String, dynamic>.from(data);
    }

    throw Exception('Invalid payment response');
  }

  Future<Map<String, dynamic>> createPayment({
    required String orderId,
    required double amount,
    required String method,
    String status = 'paid',
  }) async {
    final token = await AuthStorage.getToken();
    if (token == null || token.isEmpty) {
      throw Exception('Missing token');
    }

    final response = await _api.post(
      '/payments',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer $token',
      },
      body: '{"order_id":"$orderId","amount":$amount,"method":"$method","status":"$status"}',
    );

    final data = response['data'];
    if (data is Map<String, dynamic>) {
      return Map<String, dynamic>.from(data);
    }

    throw Exception('Invalid create payment response');
  }
}
